<a href="https://colab.research.google.com/github/MANISH-KUMAR-CODES/Exploratory-Analysis-on-International-t20-/blob/main/Exploratory_Analysis_on_International_t20.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

##<B>Analysis ON International t20

###Lets us first import all the necessary libraries of python

In [118]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [119]:
International_T20_Data = pd.read_csv('/content/drive/MyDrive/Copy of International_T20_Data.csv')

In [120]:
#lets check whats inside our data
International_T20_Data

innings  meta.data_version  \
0     [{'1st innings': {'team': 'Australia', 'delive...                0.9   
1     [{'1st innings': {'team': 'Australia', 'delive...                0.9   
2     [{'1st innings': {'team': 'Australia', 'delive...                0.9   
3     [{'1st innings': {'team': 'Hong Kong', 'delive...                0.9   
4     [{'1st innings': {'team': 'Zimbabwe', 'deliver...                0.9   
...                                                 ...                ...   
1412  [{'1st innings': {'team': 'Sri Lanka', 'delive...                0.9   
1413  [{'1st innings': {'team': 'Bangladesh', 'deliv...                0.9   
1414  [{'1st innings': {'team': 'Netherlands', 'deli...                0.9   
1415  [{'1st innings': {'team': 'Australia', 'delive...                0.9   
1416  [{'1st innings': {'team': 'Sri Lanka', 'delive...                0.9   

     meta.created  meta.revision                    info.dates info.gender  \
0      2017-02-18              2  [datetime.date(2017, 2, 17)]        male   
1      2017-02-19              2  [datetime.date(2017, 2, 19)]        male   
2      2017-02-23              1  [datetime.date(2017, 2, 22)]        male   
3      2016-09-12              1   [datetime.date(2016, 9, 5)]        male   
4      2016-06-19              1  [datetime.date(2016, 6, 18)]        male   
...           ...            ...                           ...         ...   
1412   2016-03-05              2   [datetime.date(2016, 3, 4)]        male   
1413   2016-03-08              1   [datetime.date(2016, 3, 6)]        male   
1414   2016-02-03              1   [datetime.date(2016, 2, 3)]        male   
1415   2016-09-12              1   [datetime.date(2016, 9, 6)]        male   
1416   2016-09-12              1   [datetime.date(2016, 9, 9)]        male   

     info.match_type  info.outcome.by.wickets info.outcome.winner  info.overs  \
0                T20                      5.0           Sri Lanka          20   
1                T20                      2.0           Sri Lanka          20   
2                T20                      NaN           Australia          20   
3                T20                      NaN           Hong Kong          20   
4                T20                      NaN            Zimbabwe          20   
...              ...                      ...                 ...         ...   
1412             T20                      6.0            Pakistan          20   
1413             T20                      8.0               India          20   
1414             T20                      NaN         Netherlands          20   
1415             T20                      NaN           Australia          20   
1416             T20                      4.0           Australia          20   

      ... info.outcome.by.runs info.match_type_number info.neutral_venue  \
0     ...                  NaN                    NaN                NaN   
1     ...                  NaN                    NaN                NaN   
2     ...                 41.0                    NaN                NaN   
3     ...                 40.0                    NaN                NaN   
4     ...                  2.0                    NaN                NaN   
...   ...                  ...                    ...                ...   
1412  ...                  NaN                    NaN                1.0   
1413  ...                  NaN                    NaN                NaN   
1414  ...                 84.0                    NaN                NaN   
1415  ...                 85.0                    NaN                NaN   
1416  ...                  NaN                    NaN                NaN   

     info.outcome.method info.outcome.result info.outcome.eliminator  \
0                    NaN                 NaN                     NaN   
1                    NaN                 NaN                     NaN   
2                    NaN                 NaN                     NaN   


In [121]:
#top 5 rows
International_T20_Data.head(5)

innings  meta.data_version  \
0  [{'1st innings': {'team': 'Australia', 'delive...                0.9   
1  [{'1st innings': {'team': 'Australia', 'delive...                0.9   
2  [{'1st innings': {'team': 'Australia', 'delive...                0.9   
3  [{'1st innings': {'team': 'Hong Kong', 'delive...                0.9   
4  [{'1st innings': {'team': 'Zimbabwe', 'deliver...                0.9   

  meta.created  meta.revision                    info.dates info.gender  \
0   2017-02-18              2  [datetime.date(2017, 2, 17)]        male   
1   2017-02-19              2  [datetime.date(2017, 2, 19)]        male   
2   2017-02-23              1  [datetime.date(2017, 2, 22)]        male   
3   2016-09-12              1   [datetime.date(2016, 9, 5)]        male   
4   2016-06-19              1  [datetime.date(2016, 6, 18)]        male   

  info.match_type  info.outcome.by.wickets info.outcome.winner  info.overs  \
0             T20                      5.0           Sri Lanka          20   
1             T20                      2.0           Sri Lanka          20   
2             T20                      NaN           Australia          20   
3             T20                      NaN           Hong Kong          20   
4             T20                      NaN            Zimbabwe          20   

   ... info.outcome.by.runs info.match_type_number info.neutral_venue  \
0  ...                  NaN                    NaN                NaN   
1  ...                  NaN                    NaN                NaN   
2  ...                 41.0                    NaN                NaN   
3  ...                 40.0                    NaN                NaN   
4  ...                  2.0                    NaN                NaN   

  info.outcome.method info.outcome.result info.outcome.eliminator  \
0                 NaN                 NaN                     NaN   
1                 NaN                 NaN                     NaN   
2                 NaN                 NaN                     NaN   
3                 NaN                 NaN                     NaN   
4                 NaN                 NaN                     NaN   

  info.supersubs.New Zealand  info.supersubs.South Africa  info.bowl_out  \
0                        NaN                          NaN            NaN   
1                        NaN                          NaN            NaN   
2                        NaN                          NaN            NaN   
3                        NaN                          NaN            NaN   
4                        NaN                          NaN            NaN   

   info.outcome.bowl_out  
0                    NaN  
1                    NaN  
2                    NaN  
3                    NaN  
4                    NaN  

[5 rows x 27 columns]

In [122]:
#last 5 rows
International_T20_Data.tail(5)

innings  meta.data_version  \
1412  [{'1st innings': {'team': 'Sri Lanka', 'delive...                0.9   
1413  [{'1st innings': {'team': 'Bangladesh', 'deliv...                0.9   
1414  [{'1st innings': {'team': 'Netherlands', 'deli...                0.9   
1415  [{'1st innings': {'team': 'Australia', 'delive...                0.9   
1416  [{'1st innings': {'team': 'Sri Lanka', 'delive...                0.9   

     meta.created  meta.revision                   info.dates info.gender  \
1412   2016-03-05              2  [datetime.date(2016, 3, 4)]        male   
1413   2016-03-08              1  [datetime.date(2016, 3, 6)]        male   
1414   2016-02-03              1  [datetime.date(2016, 2, 3)]        male   
1415   2016-09-12              1  [datetime.date(2016, 9, 6)]        male   
1416   2016-09-12              1  [datetime.date(2016, 9, 9)]        male   

     info.match_type  info.outcome.by.wickets info.outcome.winner  info.overs  \
1412             T20                      6.0            Pakistan          20   
1413             T20                      8.0               India          20   
1414             T20                      NaN         Netherlands          20   
1415             T20                      NaN           Australia          20   
1416             T20                      4.0           Australia          20   

      ... info.outcome.by.runs info.match_type_number info.neutral_venue  \
1412  ...                  NaN                    NaN                1.0   
1413  ...                  NaN                    NaN                NaN   
1414  ...                 84.0                    NaN                NaN   
1415  ...                 85.0                    NaN                NaN   
1416  ...                  NaN                    NaN                NaN   

     info.outcome.method info.outcome.result info.outcome.eliminator  \
1412                 NaN                 NaN                     NaN   
1413                 NaN                 NaN                     NaN   
1414                 NaN                 NaN                     NaN   
1415                 NaN                 NaN                     NaN   
1416                 NaN                 NaN                     NaN   

     info.supersubs.New Zealand  info.supersubs.South Africa  info.bowl_out  \
1412                        NaN                          NaN            NaN   
1413                        NaN                          NaN            NaN   
1414                        NaN                          NaN            NaN   
1415                        NaN                          NaN            NaN   
1416                        NaN                          NaN            NaN   

      info.outcome.bowl_out  
1412                    NaN  
1413                    NaN  
1414                    NaN  
1415                    NaN  
1416                    NaN  

[5 rows x 27 columns]

In [123]:
#Checking info
International_T20_Data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1417 entries, 0 to 1416
Data columns (total 27 columns):
 #   Column                       Non-Null Count  Dtype  
---  ------                       --------------  -----  
 0   innings                      1417 non-null   object 
 1   meta.data_version            1417 non-null   float64
 2   meta.created                 1417 non-null   object 
 3   meta.revision                1417 non-null   int64  
 4   info.dates                   1417 non-null   object 
 5   info.gender                  1417 non-null   object 
 6   info.match_type              1417 non-null   object 
 7   info.outcome.by.wickets      651 non-null    float64
 8   info.outcome.winner          1372 non-null   object 
 9   info.overs                   1417 non-null   int64  
 10  info.player_of_match         1255 non-null   object 
 11  info.teams                   1417 non-null   object 
 12  info.toss.decision           1417 non-null   object 
 13  info.toss.winner  

In [124]:
#checking descriptive summary
International_T20_Data.describe()

meta.data_version  meta.revision  info.outcome.by.wickets   info.overs  \
count       1.417000e+03    1417.000000               651.000000  1417.000000   
mean        9.000000e-01       1.370501                 6.476190    20.063514   
std         5.108829e-15       0.575198                 1.987802     1.379399   
min         9.000000e-01       1.000000                 1.000000    20.000000   
25%         9.000000e-01       1.000000                 5.000000    20.000000   
50%         9.000000e-01       1.000000                 7.000000    20.000000   
75%         9.000000e-01       2.000000                 8.000000    20.000000   
max         9.000000e-01       5.000000                10.000000    50.000000   

       info.outcome.by.runs  info.match_type_number  info.neutral_venue  
count            721.000000              684.000000               328.0  
mean              39.088766              748.774854                 1.0  
std               33.385757              190.050264                 0.0  
min                1.000000              380.000000                 1.0  
25%               13.000000              627.000000                 1.0  
50%               30.000000              746.500000                 1.0  
75%               56.000000              883.250000                 1.0  
max              249.000000             1104.000000                 1.0

In [125]:
#lets us define a function to replace column names
def replace_name(dataframe):
    dataframe.astype(str)
    dataframe_new_column=[]
    for i in dataframe.columns:
       if  'meta.' in i:
          i = i.replace('meta.','')
          dataframe_new_column.append(i)
       else: 
             'info.' in i
             i = i.replace('info.','')
             dataframe_new_column.append(i)
    return dataframe_new_column

In [126]:
new_columns = replace_name(International_T20_Data)
new_columns

['innings',
 'data_version',
 'created',
 'revision',
 'dates',
 'gender',
 'match_type',
 'outcome.by.wickets',
 'outcome.winner',
 'overs',
 'player_of_match',
 'teams',
 'toss.decision',
 'toss.winner',
 'umpires',
 'venue',
 'city',
 'outcome.by.runs',
 'match_type_number',
 'neutral_venue',
 'outcome.method',
 'outcome.result',
 'outcome.eliminator',
 'supersubs.New Zealand',
 'supersubs.South Africa',
 'bowl_out',
 'outcome.bowl_out']

In [127]:
International_T20_Data.columns = new_columns
International_T20_Data.head()

innings  data_version  \
0  [{'1st innings': {'team': 'Australia', 'delive...           0.9   
1  [{'1st innings': {'team': 'Australia', 'delive...           0.9   
2  [{'1st innings': {'team': 'Australia', 'delive...           0.9   
3  [{'1st innings': {'team': 'Hong Kong', 'delive...           0.9   
4  [{'1st innings': {'team': 'Zimbabwe', 'deliver...           0.9   

      created  revision                         dates gender match_type  \
0  2017-02-18         2  [datetime.date(2017, 2, 17)]   male        T20   
1  2017-02-19         2  [datetime.date(2017, 2, 19)]   male        T20   
2  2017-02-23         1  [datetime.date(2017, 2, 22)]   male        T20   
3  2016-09-12         1   [datetime.date(2016, 9, 5)]   male        T20   
4  2016-06-19         1  [datetime.date(2016, 6, 18)]   male        T20   

   outcome.by.wickets outcome.winner  overs  ... outcome.by.runs  \
0                 5.0      Sri Lanka     20  ...             NaN   
1                 2.0      Sri Lanka     20  ...             NaN   
2                 NaN      Australia     20  ...            41.0   
3                 NaN      Hong Kong     20  ...            40.0   
4                 NaN       Zimbabwe     20  ...             2.0   

  match_type_number neutral_venue outcome.method outcome.result  \
0               NaN           NaN            NaN            NaN   
1               NaN           NaN            NaN            NaN   
2               NaN           NaN            NaN            NaN   
3               NaN           NaN            NaN            NaN   
4               NaN           NaN            NaN            NaN   

  outcome.eliminator supersubs.New Zealand  supersubs.South Africa  bowl_out  \
0                NaN                   NaN                     NaN       NaN   
1                NaN                   NaN                     NaN       NaN   
2                NaN                   NaN                     NaN       NaN   
3                NaN                   NaN                     NaN       NaN   
4                NaN                   NaN                     NaN       NaN   

   outcome.bowl_out  
0               NaN  
1               NaN  
2               NaN  
3               NaN  
4               NaN  

[5 rows x 27 columns]

In [128]:
International_T20_Data.isnull().sum()

innings                      0
data_version                 0
created                      0
revision                     0
dates                        0
gender                       0
match_type                   0
outcome.by.wickets         766
outcome.winner              45
overs                        0
player_of_match            162
teams                        0
toss.decision                0
toss.winner                  0
umpires                     22
venue                        0
city                       189
outcome.by.runs            696
match_type_number          733
neutral_venue             1089
outcome.method            1366
outcome.result            1372
outcome.eliminator        1401
supersubs.New Zealand     1416
supersubs.South Africa    1416
bowl_out                  1415
outcome.bowl_out          1415
dtype: int64

###As we can see some columns have 1415 null values out of 1417 entries.It will be better if we drop all these columns.

In [129]:
International_T20_Data=International_T20_Data.drop(['outcome.bowl_out','bowl_out','supersubs.South Africa','supersubs.New Zealand','outcome.eliminator','outcome.method','outcome.result'], axis = 1)

In [130]:
#lets again check our nulls again
International_T20_Data.isnull().sum()

innings                  0
data_version             0
created                  0
revision                 0
dates                    0
gender                   0
match_type               0
outcome.by.wickets     766
outcome.winner          45
overs                    0
player_of_match        162
teams                    0
toss.decision            0
toss.winner              0
umpires                 22
venue                    0
city                   189
outcome.by.runs        696
match_type_number      733
neutral_venue         1089
dtype: int64

In [131]:
#lets fill all the null valeues with 0
International_T20_Data=International_T20_Data.fillna(0)

In [132]:
#lets again check our nulls again
International_T20_Data.isnull().sum()

innings               0
data_version          0
created               0
revision              0
dates                 0
gender                0
match_type            0
outcome.by.wickets    0
outcome.winner        0
overs                 0
player_of_match       0
teams                 0
toss.decision         0
toss.winner           0
umpires               0
venue                 0
city                  0
outcome.by.runs       0
match_type_number     0
neutral_venue         0
dtype: int64